types/react-dom/index.d.ts

```diff
@@ -16,7 +16,7 @@ import {
     DOMAttributes, DOMElement, ReactNode, ReactPortal
 } from 'react';
 
-export function findDOMNode(instance: ReactInstance): Element;
+export function findDOMNode(instance: ReactInstance): Element | null | Text;
```

Comment:
What is `Text`?

Reply:
I thought this was intended to be `TextNode`, but `findDOMNode` on a component that only returns a string actually produces an object that just holds a reference to the Fiber instance.

```js
ReactDOM.render(
  React.createElement(class extends React.Component {
    componentDidMount () {
      console.log(ReactDOM.findDOMNode(this))
    }
    render () {
      return 'test'
    }
  }),
  document.body.appendChild(document.createElement('div'))
)
```

<img width="426" alt="image" src="https://user-images.githubusercontent.com/73085/38530601-5ad3d96c-3ca6-11e8-8dbd-b545300aa068.png">

Reply:
Scratch all of the above; that's just the codesandbox console not printing the `@@toStringTag` or constructor name for instances. That's no ordinary object, it's an instance of the `Text` DOM constructor.

Reply:
node_modules/@types/react-dom/index.d.ts(19,72): error TS2304: Cannot find name 'Text'